augur/ancestral.py

```diff
@@ -127,16 +126,59 @@ def collect_mutations_and_sequences(tt, infer_tips=False, full_sequences=False,
 
     return {"nodes": data, "mask": mask}
 
+def run_ancestral(T, aln, root_sequence=None, is_vcf=False, full_sequences=False, fill_overhangs=False,
+                  infer_ambiguous=False, marginal=False, alphabet='nuc'):
+    tt = ancestral_sequence_inference(tree=T, aln=aln, ref=root_sequence if is_vcf else None, marginal=marginal,
+                                      fill_overhangs = fill_overhangs, alphabet=alphabet,
+                                      infer_tips = infer_ambiguous)
+
+    character_map = {}
+    for x in tt.gtr.profile_map:
+        if tt.gtr.profile_map[x].sum()==tt.gtr.n_states:
+            # TreeTime treats all characters that are not valid IUPAC nucleotide chars as fully ambiguous
+            # To clean up auspice output, we map all those to 'N'
+            character_map[x] = 'N'
+        else:
+            character_map[x] = x
+    # add reference sequence to json structure. This is the sequence with
+    # respect to which mutations on the tree are defined.
+    if root_sequence:
+        root_seq = root_sequence
+    else:
+        root_seq = tt.sequence(T.root, as_string=True)
+
+    mutations = collect_mutations_and_sequences(tt, full_sequences=full_sequences,
+                          infer_tips=infer_ambiguous, character_map=character_map, is_vcf=is_vcf)
+    if root_sequence:
+        for pos, (root_state, tree_state) in enumerate(zip(root_sequence, tt.sequence(tt.tree.root, reconstructed=infer_ambiguous, as_string=True))):
+            if root_state != tree_state:
+                mutations['nodes'][tt.tree.root.name]['muts'].append(f"{root_state}{pos+1}{tree_state}")
+
+    return {'tt': tt,
+            'root_seq': root_seq,
+            'mutations': mutations}
+
 
 def register_parser(parent_subparsers):
     parser = parent_subparsers.add_parser("ancestral", help=__doc__)
     parser.add_argument('--tree', '-t', required=True, help="prebuilt Newick")
     parser.add_argument('--alignment', '-a', help="alignment in fasta or VCF format")
+    # FIXME: these three arguments should either be all there or none
+    parser.add_argument('--annotation',
+                        help='GenBank or GFF file containing the annotation')
+    parser.add_argument('--genes', nargs='+', help="genes to translate (list or file containing list)")
+    parser.add_argument('--translations', type=str, help="translated alignments for each CDS/Gene. "
+                           "Currently only supported for fasta-input. Specify the file name via a "
+                           "template like 'my_alignment_%%GENE.fasta' where %%GENE will be replaced "
```

Comment:
@rneher When I tried to use the template with double `%%` I got an error:
```
ERROR: SequenceData: loading alignment failed... builds/21L/translations/aligned.gene.%ORF1a.fasta
```
The correct template string might currently require only a single percent sign - not sure whether this is an error in the documentation or the implementation.

When I used a single `%` things worked fine.

Reply:
percent signs need to be escaped with another percent sign.

Reply:
I see, I've never used Python pre f-strings which make use of `%` for formatting unnecessary.

Adding an `f` or `r` before the string would allow us to not need to escape. But this is a matter of taste and not blocking.